src/psyclone/tests/psyir/transformations/loop_fusion_test.py

```diff
@@ -0,0 +1,585 @@
+
+    # Check second loop
+    with pytest.raises(TransformationError) as err:
+        fuse.validate(loop1, loop2)
+    assert ("Error in LoopFuseTrans transformation. The target loop must have "
+            "four children but found: []" in str(err.value))
+
+    loop2.addchild(Literal("start", INTEGER_TYPE, parent=loop2))
+    loop2.addchild(Literal("stop", INTEGER_TYPE, parent=loop2))
+    loop2.addchild(Literal("step", INTEGER_TYPE, parent=loop2))
+    loop2.addchild(Schedule(parent=loop2))
+    loop2.loop_body.addchild(Return(parent=loop2.loop_body))
+
+    # Validation should now pass
+    fuse.validate(loop1, loop2)
+
+
+# ----------------------------------------------------------------------------
+def test_fusetrans_error_not_same_parent():
+    ''' Check that we reject attempts to fuse loops which don't share the
+    same parent '''
+
+    sch1 = Schedule()
+    sch2 = Schedule()
+    loop1 = Loop(variable=DataSymbol("i", INTEGER_TYPE), parent=sch1)
+    loop2 = Loop(variable=DataSymbol("j", INTEGER_TYPE), parent=sch2)
+    sch1.addchild(loop1)
+    sch2.addchild(loop2)
+
+    loop1.addchild(Literal("1", INTEGER_TYPE, parent=loop1))  # start
+    loop1.addchild(Literal("10", INTEGER_TYPE, parent=loop1))  # stop
+    loop1.addchild(Literal("1", INTEGER_TYPE, parent=loop1))  # step
+    loop1.addchild(Schedule(parent=loop1))  # loop body
+
+    loop2.addchild(Literal("1", INTEGER_TYPE, parent=loop2))  # start
+    loop2.addchild(Literal("10", INTEGER_TYPE, parent=loop2))  # stop
+    loop2.addchild(Literal("1", INTEGER_TYPE, parent=loop2))  # step
+    loop2.addchild(Schedule(parent=loop2))  # loop body
+
+    fuse = LoopFuseTrans()
+
+    # Try to fuse loops with different parents
+    with pytest.raises(TransformationError) as err:
+        fuse.validate(loop1, loop2)
+    assert ("Error in LoopFuseTrans transformation. Loops do not have the "
+            "same parent" in str(err.value))
+
+
+# ----------------------------------------------------------------------------
+def fuse_loops(parser=None, fortran_code=None):
+    '''Helper function that fuses the first two nodes in the given
+    Fortran code, and returns the fused Fortran code as string.
+
+    :param string fortran_code: the Fortran code to loop fuse.
+
+    :returns: String representation of fused code.
+    :rtype: str
+
+    :raises TransformationError: if the loop fuse transformation cannot
+        be applied.
+    '''
+    reader = FortranStringReader(fortran_code)
+    ast = parser(reader)
+    psy = PSyFactory("nemo").create(ast)
+    schedule = psy.invokes.get("sub").schedule
+
+    loop1 = schedule.children[0]
+    loop2 = schedule.children[1]
+    fuse = NemoLoopFuseTrans()
+    fuse.apply(loop1, loop2)
+    writer = FortranWriter()
+    return writer(schedule), schedule
+
+
+# ----------------------------------------------------------------------------
+def test_fuse_ok(parser):
+    '''This tests verifies that the valid loop fuse statements work
```

Comment:
"...that loop fusion can be successfully applied to conformant loops."

Reply:
Sound so nice.